### PR TITLE
fix: Build a correct YOLOX-nano with depthwise separable convs

### PR DESCRIPTION
**Pull request by yurkovak**
Adding new arch_param 'depthwise': False, which is needed for YOLOX-nano. When its True the following layers are replaced with depthwise-separable ones:
* `Conv`s* and blocks** in the backbone (all except `Focus`);
* Blocks and 2 last downsizing  `Conv`s in the head;
* `Conv`s in two branches in `DetectX` (all except Stem and predicting layers).

*Depthwise-separable Conv = `DepthWiseConv`
**Depthwise-separable block = a block with `DepthWiseConv` in Bottleneck

Changes in `C3` and `BottleneckCSP aren't backward-compatible, but these modules are used only in yolov5_base.py and csp_darknet53.py anyway.

### CODE REVIEW

File: src/super_gradients/training/models/detection_models/csp_darknet53.py [hunk 1]
```diff
@@ -77,14 +77,29 @@ def fuseforward(self, x):
         return self.act(self.conv(x))
 
 
+class DepthWiseConv(nn.Module):
+    # STANDARD CONVOLUTION
+    def __init__(self, input_channels, output_channels, kernel, stride, activation_type: Type[nn.Module],
+                 padding: int = None):
+        super().__init__()
+
+        self.dconv = Conv(input_channels, input_channels, kernel, stride, activation_type, padding, input_channels)
```
Comment (ofrimasad) [inline src/super_gradients/training/models/detection_models/csp_darknet53.py:86]: can you please add the arg name `groups=imput_channels`

File: src/super_gradients/training/models/detection_models/yolov5_base.py [hunk 2]
```diff
@@ -243,35 +248,39 @@ def __init__(self, arch_params):
...
 
         if arch_params.yolo_type != 'yoloX':
             self._modules_list.append(
                 Detect(num_classes, anchors, channels=[width_mult(v) for v in (256, 512, 1024)]))  # 24
         else:
             strides = anchors.stride
             self._modules_list.append(
-                DetectX(num_classes, strides, activation_type, channels=[width_mult(v) for v in (256, 512, 1024)]))  # 24
+                DetectX(num_classes, strides, activation_type, channels=[width_mult(v) for v in (256, 512, 1024)],
+                        depthwise=arch_params.depthwise))  # 24
```
Comment (ofrimasad) [inline src/super_gradients/training/models/detection_models/yolov5_base.py:283]: no need to check `arch_params.depthwise`. you already have local variable `depthwise` here

File: src/super_gradients/training/models/detection_models/yolov5_base.py [hunk 3]
```diff
@@ -414,7 +423,7 @@ def _initialize_weights(self):
             elif t is nn.BatchNorm2d:
                 m.eps = 1e-3
                 m.momentum = 0.03
-            elif t in [nn.LeakyReLU, nn.ReLU, nn.ReLU6, nn.Hardswish]:
+            elif t in [nn.LeakyReLU, nn.ReLU, nn.ReLU6, nn.Hardswish, nn.SiLU]:
```
Comment (ofrimasad) [inline src/super_gradients/training/models/detection_models/yolov5_base.py:426]: is this related to the depthwise change somehow? or a separate bugfix?

**Review by ofrimasad — COMMENTED** [review]
few minor comment inline

File: src/super_gradients/training/models/detection_models/csp_darknet53.py (hunk 1 above)
Comment (yurkovak) [reply src/super_gradients/training/models/detection_models/csp_darknet53.py]: Sure, added in d17de50

File: src/super_gradients/training/models/detection_models/yolov5_base.py (hunk 3 above)
Comment (yurkovak) [reply src/super_gradients/training/models/detection_models/yolov5_base.py:426]: Separate indeed. I was just comparing the Nano model definition in our codebase vs the source and noticed that SiLU isn't inplace, unlike in the original implementation
